tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -478,3 +533,71 @@ def _make_tf_features(self, input_feat):
                                   [-1, self._feature_size])
       freq_inputs.append(cur_input)
     return freq_inputs
+
+
+class LSTMCellBlock(rnn_cell.RNNCell):
+  """Basic LSTM recurrent network cell.
+
+  The implementation is based on: http://arxiv.org/abs/1409.2329.
+
+  We add forget_bias (default: 1) to the biases of the forget gate in order to
+  reduce the scale of forgetting in the beginning of the training.
+
+  It does not allow cell clipping, a projection layer, and does not
+  use peep-hole connections: it is the basic baseline.
+
+  Unlike BasicLSTMCell, this is a monolithic op and should be much faster. The
+  weight and bias matrixes should be compatible as long as the variabel scope
+  matches.
+  """
+
+  def __init__(self, num_units, forget_bias=1.0, input_size=None):
+    """Initialize the basic LSTM cell.
+
+    Args:
+      num_units: int, The number of units in the LSTM cell.
+      forget_bias: float, The bias added to forget gates (see above).
+    """
+    if input_size is not None:
+      logging.warn("%s: The input_size parameter is deprecated." % self)
+    self._num_units = num_units
+    self._forget_bias = forget_bias
+
+  @property
+  def state_size(self):
+    return self._num_units * 2
+
+  @property
+  def output_size(self):
+    return self._num_units
+
+  def __call__(self, x, states_prev, scope=None):
```

Comment:
Should be in the next push.
On May 11, 2016 3:02 PM, "William Chan" <notifications@github.com> wrote:

> In tensorflow/contrib/rnn/python/ops/rnn_cell.py
> <https://github.com/tensorflow/tensorflow/pull/2002#discussion_r62934796>:
>
> > +      forget_bias: float, The bias added to forget gates (see above).
> > +    """
> > +    if input_size is not None:
> > +      logging.warn("%s: The input_size parameter is deprecated." % self)
> > +    self._num_units = num_units
> > +    self._forget_bias = forget_bias
> > +
> > +  @property
> > +  def state_size(self):
> > +    return self._num_units * 2
> > +
> > +  @property
> > +  def output_size(self):
> > +    return self._num_units
> > +
> > +  def __call__(self, x, states_prev, scope=None):
>
> is this on the public github? dont see any examples for list for
> state_size.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/2002/files/fc40971bba82762fe413b5e7d6fe12a09722876b#r62934796>
>